genie_registry/clinical.py

```diff
@@ -175,6 +175,37 @@ def _check_int_year_consistency(
     return ""
 
 
+def _check_year_death_validity(clinicaldf: pd.DataFrame) -> str:
+    """
+    Check if YEAR_DEATH >= YEAR_CONTACT.
+    YEAR_DEATH should alway be equal or large than YEAR_CONTACT if they are both available.
+    This function compares rows with numeric values in both columns and returns comparion results("True"/"False")
+    If either of the column contains NA or nominal data (e.g. "Unknown", "Not Collected", "Unknown", "Not Applicable"),
+    then "N/A" will be outputed.
+
+    Args:
+        clinicaldf: Clinical Data Frame
+
+    Returns:
+        Error message if YEAR_DEATH if invalid
+    """
+    error = ""
+    # Generate temp dataframe to handle datatype mismatch in a column
+    temp = clinicaldf.copy()
```

Comment:
Nit: to reduce the amount of memory used, you may want to scope this to copy only the two columns in question.

You're also going to have another task with `INT_DOD` >= `INT_CONTACT`. No need to change anything here, but you may want to when you tackle the other function so you don't duplicate code.  Minor detail though.